### PR TITLE
Signup Page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Signup Page

This PR has signup page components, signup form validation and theme toggle.

- Files to review:
  - [ ] src/components/Signup/Signup.js
  - [ ] src/components/Signup/SignupCard.js
  - [ ] src/components/Signup/SignupForm.js
  - [ ] src/components/Signup/SignupHeader.js
  - [ ] src/contexts/theme.context.js
  - [ ] src/utils/inputValidation.util.js
  - [ ] src/utils/regExp.util.js
  - [ ] src/utils/theme.util.js
  - [ ] src/utils/timer.util.js

- Use this link for preview - https://deploy-preview-2--pensive-perlman-311072.netlify.app/

- Screenshots:
d-1
d-2

Thanks,

### CODE REVIEW

File: src/components/Reusable/Button.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+const Button = (props) => {
+
+    return (
+
+        <button onClick={props.onClick} className={props.classes}>
```
Comment (nayyyhaa) [inline src/components/Reusable/Button.js]: you can destructure props here.

File: src/components/Reusable/Card.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { useTheme } from "../../contexts/theme.context"
+
+const Card = (props) => {
+    const { theme } = useTheme()
+
+    return (
+
+        <article className={`${props.classes} ${theme === 'dark' && 'b-solid b-secondary'} card-dim card-shadow-xs`}>
```
Comment (nayyyhaa) [inline src/components/Reusable/Card.js]: good use of conditional class names here.

File: src/components/Reusable/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import Form from "./Form";
+import Header from "./Header";
+import Icon from "./Icon";
+import Input from "./Input";
+import Label from "./Label";
+import Main from "./Main";
+import NavBar from "./NavBar";
+import Text from "./Text";
+
+export {
```
Comment (nayyyhaa) [inline src/components/Reusable/index.js]: you can directly export components, no need to import + export here

**Review by nayyyhaa — APPROVED** [review]
Suggested a few changes. Loved that you've used reusable components like `input` & `labels`. Nice work.

File: src/components/Signup/SignupForm.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import { Form, Input, Label, Button, Text } from "../Reusable"
+import { getSolidBtnTextColor, getSolidBtnBgColor, timer, emailIsInvalid, nameIsInvalid, passIsInvalid, passAndConfPassAreDiff } from "../../utils"
+import { useTheme } from '../../contexts/theme.context'
+
+const SignupForm = () => {
+    const [enteredEmail, setEnteredEmail] = useState('')
+    const [emailError, setEmailError] = useState(false)
+
+    const [enteredFName, setEnteredFName] = useState('')
+    const [fNameError, setFNameError] = useState(false)
```
Comment (vrx29) [inline src/components/Signup/SignupForm.js:11]: You can use reducer here instead of state.

File: src/components/Signup/SignupForm.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                    show password
+                </Label>
+
+            </div>
+
+            <div className='flx flx-maj-start flx-min-center mg-btm-s'>
+
+                <Input type='checkbox' id='terms-check' />
+
+                <Label for='terms-check' classes={`txt-md txt-cap mg-left-xs`}>
```
Comment (vrx29) [inline src/components/Signup/SignupForm.js:115]: try to put input element inside label

**Review by vrx29 — APPROVED** [review]

File: src/components/Reusable/Button.js (hunk 1 above)
Comment (foolhardy21) [reply src/components/Reusable/Button.js]: made the changes. thanks.

File: src/components/Reusable/Card.js (hunk 2 above)
Comment (foolhardy21) [reply src/components/Reusable/Card.js]: thank you.

File: src/components/Reusable/index.js (hunk 3 above)
Comment (foolhardy21) [reply src/components/Reusable/index.js]: made the changes. thanks for pointing out.

File: src/components/Signup/SignupForm.js (hunk 4 above)
Comment (foolhardy21) [reply src/components/Signup/SignupForm.js:11]: thanks for pointing out. i have made that change in another pr.

File: src/components/Signup/SignupForm.js (hunk 5 above)
Comment (foolhardy21) [reply src/components/Signup/SignupForm.js:115]: will make the change. thanks.

**Pull request merged**